File: app/src/main/java/com/polestar/charging/ui/cluster/view/renderer/MarkerWithPosition.java
package com.polestar.charging.ui.cluster.view.renderer;

import com.amap.api.maps.model.LatLng;
import com.amap.api.maps.model.Marker;

/**
 * A Marker and its position. {@link Marker#getPosition()} must be called from the UI thread, so this
 * object allows lookup from other threads.
 */
public class MarkerWithPosition {
    private final Marker marker;
    private LatLng position;

    public MarkerWithPosition(Marker marker) {
        this.marker = marker;
        position = marker.getPosition();
    }

    @Override
    public boolean equals(Object other) {
        if (other instanceof MarkerWithPosition) {
            return marker.equals(((MarkerWithPosition) other).marker);
        }
        return false;
    }

    public void setPosition(LatLng position) {
        this.position = position;
    }

    public LatLng getPosition() {
        return position;
    }

    public Marker getMarker() {
        return marker;
    }

    @Override
    public int hashCode() {
        return marker.hashCode();
    }
}

File: app/src/main/java/com/polestar/charging/ui/cluster/MarkerWithPosition.kt
package com.polestar.charging.ui.cluster

import com.amap.api.maps.model.Marker

class MarkerWithPosition(val marker: Marker) {
    var position = marker.position

    override fun equals(other: Any?): Boolean {
        if (other is MarkerWithPosition) {
            return marker.equals(other.marker)
        }
        return false
    }

    override fun hashCode(): Int {
        return marker.hashCode()
    }
}